### PR TITLE
Github workflow action for building & releasing Bitshares beet

### PR DESCRIPTION
**Pull request by grctest**
Still a WIP, the output setup results in the following error after launching the app post install: `cannot find module '@babel/runtime/helpers/asyncToGenerator'`

I've tried installing the @babel/runtime package but this doesn't change the error.

The workflow focuses on the windows build for now, however once fully operational the next priority is suppporting linux/macos.

### CODE REVIEW

**Comment by abitmore** [comment]
> Still a WIP, the output setup results in the following error after launching the app post install: `cannot find module '@babel/runtime/helpers/asyncToGenerator'`

I found this: https://github.com/yarnpkg/yarn/issues/8054#issuecomment-634153330:
>`export NODE_PATH=/usr/lib/nodejs:/usr/share/nodejs`
>
>https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=933229

**Comment by grctest** [comment]
Cool, that's github actions successfully building windows setup: https://github.com/grctest/beet/releases/tag/v1.0.3

**Comment by grctest** [comment]
Looks like appveyor still wants to run without the appveyor file?

**Comment by sschiessl-bcp** [comment]
Looks also like it is (sometimes?) working https://ci.appveyor.com/project/clockworkgr/beet/builds/44753286/artifacts

Possible we need to tun the link off from AppVeyor end too...

**Comment by grctest** [comment]
As far as I can tell the action workflow has successfully built and packaged the linux deb + appimage, as well as macos dmg. https://github.com/grctest/beet/releases/tag/v1.0.9 (not tested running these new packages on their respective os yet)

**Comment by grctest** [comment]
RE `uses: BTS-CM/action-gh-release@cd28b0f5ee8571b76cfdaa62a30d51d752317477` I forked the action-gh-release so as to freeze the state of the workflow action to that commit, reducing risk of supply chain attack.

**Comment by grctest** [comment]
I've successfully implemented virustotal scanning of deliverables in another repo, should we add this to beet too? In order to establish presence as a safe application in all/most antivirus apps. https://github.com/BTS-CM/NFT_Viewer/releases/tag/v1.0.26  https://github.com/BTS-CM/NFT_Viewer/blob/main/.github/workflows/node.js.yml

**Comment by sschiessl-bcp** [comment]
Did you see the changelist actions https://github.com/bitshares/bitshares-ui/blob/c0a2774f8703b23743c97fe159f19f946770e497/.github/workflows/build-release-binaries.yml#L43, and what is the reason to use a custom release actions instead of actions/create-release?

**Comment by grctest** [comment]
> what is the reason to use a custom release actions instead of actions/create-release

actions/create-release is [unmaintained & archived](https://github.com/actions/create-release#github-action---releases-api), they recommended https://github.com/softprops/action-gh-release which I then forked to freeze our use to a specific commit to reduce risk of potential supply chain attack, we could use a bitshares org forked repo or always trust the latest softprops repo

**Comment by grctest** [comment]
> Did you see the changelist actions

Do you want to use that action repo for changelogs? What do you think of this action instead? https://github.com/marketplace/actions/release-changelog-builder

**Comment by grctest** [comment]
@sschiessl-bcp Any further thoughts on the above?

File: src/store/modules/AccountStore.js [hunk 1]
```diff
@@ -33,28 +33,27 @@ const actions = {
         state
     }, payload) {
         return new Promise((resolve, reject) => {
-            let index = -1;
-            for (let i = 0; i < state.accountlist.length; i++) {
-                if (payload.account.chain == state.accountlist[i].chain && payload.account.accountID == state.accountlist[i].accountID) {
-                    reject('Account already exists');
+            let existingAccount = state.accountlist.find(x => x.chain == payload.account.chain && x.accountID == payload.account.accountID)
```
Comment (sschiessl-bcp) [inline src/store/modules/AccountStore.js:36]: What is this change about?

**Review by sschiessl-bcp — CHANGES_REQUESTED** [review]

File: src/store/modules/AccountStore.js (hunk 1 above)
Comment (grctest) [reply src/store/modules/AccountStore.js:36]: IIRC the reject wasn't triggering within the loop so I removed the loop, kept the logic and now the account adding bug is gone 👍

Comment (sschiessl-bcp) [reply src/store/modules/AccountStore.js:36]: Ok, so its a change that is unrelated to this PR

**Comment by sschiessl-bcp** [comment]
> @sschiessl-bcp Any further thoughts on the above?

Not at the moment

**Pull request merged**